cql3/expr/restrictions.cc

```diff
@@ -143,6 +166,19 @@ void preliminary_binop_vaidation_checks(const binary_operator& binop) {
             throw exceptions::invalid_request_exception("LIKE cannot be used for Multi-column relations");
         }
 
+        if(binop.op == oper_t::IN) {
+            auto valid_size = lhs_tup->elements.size();
+            if (is<collection_constructor>(binop.rhs)) {
+                const std::vector<expression>& elements = as<collection_constructor>(binop.rhs).elements;
+                validate_tuples_size (elements, valid_size);                
+            } else if (auto rhs_const = as_if<constant>(&binop.rhs)) {
```

Comment:
I'm not familiar with those weird objects (collection constructor, constant, etc.) so @avikivity better review this code and approve it.
But just one question - are both cases of this if() covered by the test cases we already had?
By the way, as usual, while you were writing this code and were thinking of various edge case, maybe you thought of additional edge cases that need to be tested. If you did, please add them to the tests (but not in the cassandra_tests/ directory - in the other files).

Reply:
@nyh "are both cases of this if() covered by the test cases we already had?" I hope so. I had to write this if because the tests didn't pass without it.